up-l1/README.adoc

```diff
@@ -311,6 +318,34 @@ end
 deactivate T
 ----
 
+== LocalUriProvider
+
+A uEntity can use the `LocalUriProvider` to create URIs representing the uEntity's local resources during runtime. This information can then be used in messages to be sent to other uEntities.
+
+A `UTransport` implementation can use the `LocalUriProvider` to determine the uEntity's authority during runtime. This information can might be useful for normalizing _local_ URIs passed into the Transport Layer API methods with authority information.
```

Comment:
Now I asked users to pass uEntity's authority while initializing Zenoh UTransport. Then I can use the authority to form UUri. Since we will have LocalUriProvider for UTransport to use, do you think we should have a consistent way to initialize uAuthority and uEntity? Or perhaps we should leave the liberty to developers to decide how they get uAuthority and uEntity?

Reply:
The idea is that your transport library also provides an implementation of `LocalUriProvider`. It is up to you, how the authority and entity ID are being initialized, e.g. by means of passing them into a `new` function or reading some environment variables etc. You seem to have chosen the former approach for your UTransport implementation, so why not use the same approach for `LocalUriProvider`? You can then pass a reference to `LocalUriProvider` into your UTransport implementation ...

Reply:
Makes sense to me